srcdeps-core/src/main/java/org/srcdeps/core/GavSet.java

```diff
@@ -102,10 +183,29 @@ public Builder includes(String... rawPatterns) {
 
     }
 
+    private static final List<GavPattern> EMPTY_LIST = Collections.emptyList();
+    private static final GavSet INCLUDE_ALL = new GavSet(Collections.singletonList(GavPattern.matchAll()), EMPTY_LIST);
+
+    private static void append(List<GavPattern> cludes, Appendable out) throws IOException {
+        boolean first = true;
+        for (GavPattern gavPattern : cludes) {
+            if (first) {
```

Comment:
I think you can just leave this outside of the `if` and let the VM inline this call.

Reply:
> I think you can just leave this outside of the if

Could you please write down how do you mean it?

Reply:
Something like:

```java
for (GavPattern gavPattern : cludes) {
  if (!first) {
    out.append(',');
  }
  out.append(gavPattern.toString());
  first = false;
}
```

Reply:
Interesting. Which  JIT optimization technique is supposed to be applicable in your version (and not applicable in my version)? Branch prediction? Or maybe loop unrolling? Or you mean that the whole method can be inlined?

Reply:
I guess you are right, the prediction will be the same. On my version, the assignment will be a noop, but you have it only on the one-time-only branch. So, I guess we are then talking only about code readability :)